### PR TITLE
Add deployment script for katana contracts

### PR DESCRIPTION
**Pull request by Anyhowclick**
- Script deploys staking, governance, voting power strategy and executors contracts

### CODE REVIEW

File: deployment/katanaDeployment.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+let kncAddress;
+let deployerAddress;
+let epochPeriod;
+let starttime;
+let shortExecutorConfig;
+let longExecutorConfig;
+let daoOperator;
+let outputFilename;
+
+task('deployGovInfra', 'deploys staking, governance, voting power strategy and executors').setAction(async () => {
```
Comment (ducquangkstn) [inline deployment/katanaDeployment.js:36]: Pls config gas price here

File: deployment/katanaDeployment.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      'staking': kyberStaking.address,
+      'governance': kyberGovernance.address,
+      'shortExecutor': shortExecutor.address,
+      'longExecutor': longExecutor.address,
+      'votingPowerStrategy': votingPowerStrategy.address
+  });
+
+  // set executors and voting power strategy in governance
+  console.log(`authorize executors...`);
+  await kyberGovernance.authorizeExecutors([shortExecutor.address, longExecutor.address]);
```
Comment (ducquangkstn) [inline deployment/katanaDeployment.js:109]: This should be in constructor, right?

File: deployment/katanaDeployment.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  console.log(`longExecutor address: ${longExecutor.address}`);
+  await pressToContinue();
+
+  console.log('deploying voting power strat...');
+  const VotingPowerStrategy = await ethers.getContractFactory('EpochVotingPowerStrategy');
+  const votingPowerStrategy = await VotingPowerStrategy.deploy(
+    kyberGovernance.address,
+    kyberStaking.address
+  );
+  await votingPowerStrategy.deployed();
```
Comment (ducquangkstn) [inline deployment/katanaDeployment.js:95]: Well actually, you can await multiple contracts are deployed at the same time, by using `Promise.all`

File: deployment/katanaDeployment.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+let outputFilename;
+
+task('deployGovInfra', 'deploys staking, governance, voting power strategy and executors').setAction(async () => {
+  parseInput(configParams);
+  const [deployer] = await ethers.getSigners();
+  deployerAddress = await deployer.getAddress();
+
+  // contract deployment
+  console.log('deploying staking contract...');
+  const KyberStaking = await ethers.getContractFactory('KyberStaking');
```
Comment (ducquangkstn) [inline deployment/katanaDeployment.js:43]: we need to staking.updateWithdrawHandler to voitingPowerStrategy

**Review by ducquangkstn — CHANGES_REQUESTED** [review]

File: deployment/katanaDeployment.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+  // set executors and voting power strategy in governance
+  console.log(`authorize executors...`);
+  await kyberGovernance.authorizeExecutors([shortExecutor.address, longExecutor.address]);
+  console.log(`authorize voting power strategy...`);
+  await kyberGovernance.authorizeVotingPowerStrategies([votingPowerStrategy.address]);
+
+  // transfer admin to governance
+  console.log('transfer staking admin rights to governance...');
+  await kyberStaking.transferAdminQuickly(kyberGovernance.address);
```
[the remark this answers is not in the document]
Comment (Anyhowclick) [reply deployment/katanaDeployment.js:115]: right, yes

File: deployment/katanaDeployment.js (hunk 2 above)
Comment (Anyhowclick) [reply deployment/katanaDeployment.js:109]: circular dependency. executors need governance in ctor, governance needs executors in ctor.

File: deployment/katanaDeployment.js (hunk 3 above)
Comment (Anyhowclick) [reply deployment/katanaDeployment.js:95]: that is true, but from mainnet experience, better to do and monitor 1 by 1, gas price fluctuates very quickly.. maybe should ping user for input for gas price to be used for the next tx

File: deployment/katanaDeployment.js (hunk 4 above)
Comment (Anyhowclick) [reply deployment/katanaDeployment.js:43]: ok thanks

File: deployment/katanaDeployment.js (hunk 3 above)
Comment (Anyhowclick) [reply deployment/katanaDeployment.js]: updated to be as such

**Review by ducquangkstn — APPROVED** [review]

**Pull request merged**